src/goto-programs/goto_inline_class.cpp

```diff
@@ -367,7 +368,7 @@ void goto_inlinet::insert_function_body(
   t_it=goto_function.body.instructions.begin();
   unsigned begin_location_number=t_it->location_number;
   t_it=--goto_function.body.instructions.end();
-  assert(t_it->is_end_function());
+  DATA_INVARIANT(t_it->is_end_function(), "end of function");
```

Comment:
If possible could the message include a short explanation of why this should be impossible?